### PR TITLE
Load cookbook versions correctly for knife

### LINKED ISSUE #4934: 'knife cookbook show' broken in 12.9.41... undefined method `preferred_manifest_record'
Using Chef: `12.9.41` when i try to show a cookbook files contents with `knife cookbook show <name> <version> <file_name>` i get the below error:

```
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/knife/cookbook_show.rb:72:in `run': undefined method `preferred_manifest_record' for #<Hash:0x00000002ce3db0> (NoMethodError)
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/knife.rb:421:in `block in run_with_pretty_exceptions'
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/local_mode.rb:44:in `with_server_connectivity'
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/knife.rb:420:in `run_with_pretty_exceptions'
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/knife.rb:219:in `run'
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/lib/chef/application/knife.rb:148:in `run'
        from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.9.41/bin/knife:25:in `<top (required)>'
        from /usr/bin/knife:51:in `load'
        from /usr/bin/knife:51:in `<main>'
```

Looking at `cookbook_show.rb:72` it's trying to call `preferred_manifest_record` on the returned cookbook version `Hash` object and fails.

```
cookbook = rest.get("cookbooks/#{cookbook_name}/#{cookbook_version}")
manifest_entry = cookbook.preferred_manifest_record(node, segment, filename)
```

### PR DESCRIPTION
**Pull request by thommay**
Fixes #4934

### CODE REVIEW

File: lib/chef/knife/cookbook_show.rb [hunk 1]
```diff
@@ -51,6 +51,11 @@ class CookbookShow < Knife
         :description => "Show corresponding URIs"
 
       def run
+        cookbook_name, segment, filename = @name_args[0], @name_args[2], @name_args[3]
+        cookbook_version = @name_args[1] == "latest" ? "_latest" : @name_args[1]
```
Comment (jkeiser) [inline lib/chef/knife/cookbook_show.rb]: Will not hold this up for it, and it's totally up to your judgement, but I find avoiding the ternary operator where possible to make things more readable:

```
cookbook_version = @name_args[1]
cookbook_version = "_latest" if cookbook_version == "latest"
```

**Comment by jkeiser** [comment]
:+1:, one comment you may deal with (or not) as you see fit.

File: lib/chef/knife/cookbook_show.rb (hunk 1 above)
Comment (thommay) [inline lib/chef/knife/cookbook_show.rb]: actually we can do better, because `CookbookVersion.load` does the coercion for `latest` anyway.

**Comment by lamont-granquist** [comment]
👍

**Pull request merged**